x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PushDownJoinPastProjectTests.java

```diff
@@ -240,6 +240,137 @@ public void testShadowingAfterPushdown2() {
         assertLeftJoinConfig(join.config(), "emp_no", mainRel.outputSet(), "languages", lookupRel.outputSet());
     }
 
+    /**
+     * Project[[languages{f}#30, emp_no{f}#27, salary{f}#32]]
+     * \_Limit[1000[INTEGER],true]
+     *   \_Join[LEFT,[emp_no{f}#16, languages{f}#30],[emp_no{f}#16],[languages{f}#30],emp_no{f}#16 == languages{f}#30]
+     *     |_Limit[1000[INTEGER],false]
+     *     | \_EsRelation[test][_meta_field{f}#22, emp_no{f}#16, first_name{f}#17, ..]
+     *     \_EsRelation[test_lookup][LOOKUP][emp_no{f}#27, languages{f}#30, salary{f}#32]
+     */
+    public void testShadowingAfterPushdownExpressionJoin() {
+        assumeTrue("Requires LOOKUP JOIN", EsqlCapabilities.Cap.JOIN_LOOKUP_V12.isEnabled());
+
+        String query = """
+            FROM test_lookup
+            | RENAME languages as lang2
+            | EVAL y = emp_no
+            | RENAME y AS lang
+            | LOOKUP JOIN test_lookup ON lang == languages
+            | KEEP languages, emp_no, salary
+            """;
+
+        var plan = optimizedPlan(query);
+        var project = as(plan, Project.class);
+        var limit1 = asLimit(project.child(), 1000, true);
+        var join = as(limit1.child(), Join.class);
+        var lookupRel = as(join.right(), EsRelation.class);
+        var limit2 = asLimit(join.left(), 1000, false);
+        var mainRel = as(limit2.child(), EsRelation.class);
+
+        var projections = project.projections();
+        assertThat(Expressions.names(projections), contains("languages", "emp_no", "salary"));
+
+        var languages = as(projections.get(0), FieldAttribute.class);
+        assertEquals("languages", languages.fieldName().string());
+        assertTrue(lookupRel.outputSet().contains(languages));
```

Comment:
++, this is subtle but it's important that you use an `AttributeSet` here, because the name id matters.